### PR TITLE
Allow embedded bindings and message attachments to be shown simultaneously

### PR DESCRIPTION
**Pull request by mickmister**
#### Summary

This PR makes it so both the `app_bindings` and `attachments` can be used simultaneously. Before this PR, the bindings would take precedence. Now the bindings are rendered just above the attachments.

#### Ticket Link

Fixes https://mattermost.atlassian.net/browse/MM-45785

#### Screenshots

Before

image

After

image

#### Release Note

```release-note
App bindings now allow embedded forms and message attachments simultaneously
```

### CODE REVIEW

**Comment by mickmister** [comment]
I believe this change needs to be ported to mobile. @jfrerich What's the process of making this change with v2?

**Review by larkox — APPROVED** [review]

**Comment by larkox** [comment]
Regarding the mobile related changes, the general idea is that the team adding the feature, handles the future both in Web and Mobile. Not sure if there is any other discussion in place regarding this particular topic.

**Comment by mickmister** [comment]
@larkox I made a ticket for mobile here https://mattermost.atlassian.net/browse/MM-47015

> the general idea is that the team adding the feature, handles the feature both in Web and Mobile

When you say "adding the feature", do you mean only new features? Like will embedded forms in general be implemented by the integrations team? That's fine with me, I just need to know when is the right time is to do this, and probably some hand-holding to know how to fit it into the new system.

**Comment by larkox** [comment]
The only thing that V2 is missing to start adding this is actually fetch and store the bindings (so it can be properly tested).

Regarding hand-holding, feel free to reach to the mobile team in https://community.mattermost.com/core/channels/ship-gekidou--mobile-v2 . We are more than happy to help you with anything you need :smile:

File: components/post_view/post_body_additional_content/post_body_additional_content.tsx [hunk 1]
```diff
@@ -149,22 +149,26 @@ export default class PostBodyAdditionalContent extends React.PureComponent<Props
...
     }
 
+    renderAppBindings = () => {
+        return (
+            <EmbeddedBindings
+                embeds={this.props.post.props.app_bindings}
+                post={this.props.post}
+            />
+        );
+    }
```
Comment (jfrerich) [inline components/post_view/post_body_additional_content/post_body_additional_content.tsx:159]:
```suggestion
    renderAppBindings = () => (
        <EmbeddedBindings
            embeds={this.props.post.props.app_bindings}
            post={this.props.post}
        />
    );
```

**Review by jfrerich — APPROVED** [review]
LGTM with one NIT

File: components/post_view/post_body_additional_content/post_body_additional_content.tsx (hunk 1 above)
Comment (mickmister) [reply components/post_view/post_body_additional_content/post_body_additional_content.tsx:159]: Leaving it as-is to match other functions in this file

**Comment by mickmister** [comment]
@hanzei: @asatkinson and I talked about this PR, and we think the solution here is sub-optimal. Ideally the frontend should be able to combine attachments and embedded bindings, to use the stylings of slack attachments, while getting the form submission logic from the App framework.

In this PR, they're currently rendered as entirely different entities. The embedded bindings aren't so "embedded". I think we can make it so the slack attachment struct itself supports rendering bindings, just as it would with its usual interactive forms.

**Comment by hanzei** [comment]
1/5 the PR is a step in the right direction. Surely, we could make slack attachments and embedded bindings play nicely together, but that is not going to happen soon, right?

**Comment by hanzei** [comment]
@mickmister With that in mind, is this PR ready for review?

**Comment by mickmister** [comment]
@hanzei Yes this is ready for review cc @DHaussermann

**Comment by mickmister** [comment]
@hanzei Looks like we'll have to migrate with the monorepo https://community.mattermost.com/private-core/pl/k6fd5qddoibyfedczg4r5rdnuo

**Review by DHaussermann — APPROVED** [review]
Tested and passed
- No issues with both slack attachment and embedded when both are used
- No issue when using only embedded and attachment
- Based on message content the 1st link with OpenGraph data is shown in the preview.

LGTM!

Thanks to @mickmister for helping with test tooling on this to create the posts!

**Comment by M-ZubairAhmed** [comment]
Hello there! Could you please open the PR in the mattermost-server as part of the monorepo? I will be closing this PR for now. Thank you very much for your understanding and cooperation in this matter.

**Pull request closed without merging**